### PR TITLE
Remove obsolete inbox messages

### LINKED ISSUE #4153: Remove inbox message: WC-Admin Welcome message
Remove Welcome message about WC-Admin: 

This should also be removed from the inbox of existing users.

### LINKED ISSUE #4154: Remove inbox message: review your experience
Remove inbox message about reviewing wc-admin plugin:

This should also be removed in existing users

### LINKED ISSUE #4155: Remove inbox message: Store notice setting
Remove inbox message about store notice setting

It should also be removed from the inbox of existing users that already received the message.

### PR DESCRIPTION
**Pull request by mattsherman**
Fixes #4153, Fixes #4154, Fixes #4155

The following inbox messages are no longer shown to users, as they no longer are needed in the onboarding experience:
* “New feature(s)”
* “Looking for the Store Notice setting”
* “Review your experience”

### Screenshots
These inbox messages will no longer be shown:

#### New installation

* Run this branch
* Perform a new installation of WooCommerce and WooCommerce Admin (make sure you activate woocommerce-admin plugin)
* With WP-CLI, update the woocommerce_admin_install_timestamp option and force the wc_admin_daily cron to run, so that the code that would have added the “Review your experience” message is run (if normally only shows the message after 3 days).

```
wp option update woocommerce_admin_install_timestamp 0
wp cron event run wc_admin_daily
```

* Verify that obsolete inbox messages do not appear

#### Upgrade

* Install WooCommerce and WooCommerce Admin

* With WP-CLI, update the woocommerce_admin_version option so that the WooCommerce Admin plugin thinks it needs to upgrade

```
wp option update woocommerce_admin_version 0.1
```

* Run this branch

* Perform an upgrade of WooCommerce Admin (make sure you activate woocommerce-admin plugin)

* With WP-CLI, update the woocommerce_admin_install_timestamp option and force the wc_admin_daily cron to run, so that the code that would have added the “Review your experience” message is run (if normally only shows the message after 3 days).

```
wp option update woocommerce_admin_install_timestamp 0
wp cron event run wc_admin_daily
```

* Verify that obsolete inbox messages do not appear
### Unit test instructions:

```
./vendor/bin/phpunit tests/notes/class-wc-tests-notes-data-store.php
```

### Changelog Note:

Tweak: Remove obsolete inbox onboarding messages.

### CODE REVIEW

File: src/Install.php [hunk 1]
```diff
@@ -448,16 +449,28 @@ public static function create_events() {
...
-		// @todo This is potentially redundant when the core package exists.
+		// Note: this is potentially redundant when the core package exists.
 		wp_schedule_single_event( time() + 10, 'generate_category_lookup_table' );
 	}
 
+	/**
+	 * Delete obsolete notes.
+	 */
+	protected static function delete_obsolete_notes() {
+		$obsolete_notes_names = array(
```
Comment (joshuatf) [inline src/Install.php:460]: Do you think there is any point in adding a filter here for other packages/plugins to hook into?

File: tests/notes/class-wc-tests-notes-data-store.php [hunk 2]
```diff
@@ -84,4 +85,117 @@ public function test_read_with_invalid_content_data() {
 		$this->assertEquals( 'PHPUNIT_TEST_NOTE', $note->get_title() );
 		$this->assertEquals( 'PHPUNIT_TEST_NOTE_CONTENT', $note->get_content() );
 	}
+
+	/**
+	 * Tests that the delete_notes_with_name method works when a single name is
+	 * passed in.
+	 */
+	public function test_delete_notes_with_single_name() {
```
Comment (joshuatf) [inline tests/notes/class-wc-tests-notes-data-store.php:93]: ❤️ Thanks for the tests.

**Review by joshuatf — APPROVED** [review]
Bumped the version number to simulate an update and old notes removed.  Testing well for me and LGTM pending one question.

File: src/Install.php (hunk 1 above)
Comment (mattsherman) [reply src/Install.php:460]: Yeah, that would be a nice addition. I'll add a filter to this PR.

Comment (mattsherman) [reply src/Install.php:460]: Added filter.

**Comment by mattsherman** [comment]
@joshuatf Thanks for the review and the suggestion of the filter. I've added a filter (not really sure how to best test this without manually updating code... is there a file in a WP/WooCommerce install that is meant for one-off/test things like this to be temporarily added?).

Ready for re-review!

**Comment by timmyc** [comment]
> is there a file in a WP/WooCommerce install that is meant for one-off/test things like this to be temporarily added?).

I took @rrennick's idea of having an odds-and-ends plugin in my local test site where I can add in random code like testing out a filter with some `error_log` statements myself.

**Comment by jeffstieler** [comment]
> is there a file in a WP/WooCommerce install that is meant for one-off/test things like this to be temporarily added?).

I just add filters and actions in my `wp-config.php` file after `require_once(ABSPATH . 'wp-settings.php');` - hacky I know, but it works well.

**Comment by rrennick** [comment]
> is there a file in a WP/WooCommerce install that is meant for one-off/test things like this to be temporarily added?

All my installs have a `wp-content/mu-plugins/test.php`. I use that because mu-plugins are loaded before plugins and I can be certain my hooks/filters are setup before any plugin filter/hook runs.

**Comment by rrennick** [comment]
> Ready for re-review!

@mattsherman 🚢

File: src/Install.php [hunk 3]
```diff
@@ -463,6 +463,18 @@ protected static function delete_obsolete_notes() {
 			'wc-admin-store-notice-giving-feedback',
 		);
 
+		$additional_obsolete_notes_names = apply_filters(
+			'woocommerce_admin_obsolete_notes_names',
```
Comment (joshuatf) [inline src/Install.php:467]: Not a necessity, but we could directly filter the notes from above so we don't need to merge multiple arrays.  I generally favor providing more control than less.

**Review by joshuatf — APPROVED** [review]
Echoing what others have said, I also have a test plugin to throw random actions/filters in for testing.

And while it's certainly not a blocker to get this merged, we have done some automated testing around filters in other areas.  https://github.com/woocommerce/woocommerce-admin/blob/e2ef363d3368a420c5085e96416136581860d9da/tests/api/onboarding-profile.php#L126

Left one optional comment, but otherwise this LGTM.

**Comment by mattsherman** [comment]
@timmyc @jeffstieler @rrennick @joshuatf Thanks for the info on how to test random actions/filters! 🙌

File: src/Install.php (hunk 3 above)
Comment (mattsherman) [reply src/Install.php:467]: By doing it this way, we ensure that no other code can override our removing of these notes, which we definitely want gone. If any filter handler didn't merge the passed in array with whatever they wanted removed, our notes would not be removed. Of course, this same basic problem would exist with any other filter handler wanted removed. It's a general brittleness of this approach.

We could revisit this in a future PR if wanted.

**Comment by mattsherman** [comment]
> And while it's certainly not a blocker to get this merged, we have done some automated testing around filters in other areas.
> 
> https://github.com/woocommerce/woocommerce-admin/blob/e2ef363d3368a420c5085e96416136581860d9da/tests/api/onboarding-profile.php#L126

Thanks for letting me know -- I'll definitely keep this in mind in for future PRs and testing!

**Pull request merged**